### PR TITLE
Use github workflow to build sdist/wheel and upload.

### PR DESCRIPTION
**Pull request by Carreau**
Setup a github workflow based on numcodec to build sdist and wheel. Adapted from numcodec. 
I have to check what ciwheel do, as zarr wheels are likely universal and dont' need to be per-platform

TODO:
* [ ] Add unit tests and/or doctests in docstrings
* [ ] Add docstrings and API docs for any new/modified user-facing classes and functions
* [ ] New/modified features documented in docs/tutorial.rst
* [ ] Changes documented in docs/release.rst
* [ ] AppVeyor and Travis CI passes
* [ ] Test coverage is 100% (Coveralls passes)

### CODE REVIEW

**Comment by Carreau** [comment]
I switched from cibuildwheel to "build" which is a non-official pep518 packed (hence adding the Pyproject.toml file as well).

**Comment by jakirkham** [comment]
Do we need to do anything special to indicate the wheel is not architecture dependent?

File: .github/workflows/releases.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+          path: dist
+      - uses: actions/download-artifact@v1
+        with:
+          name: wheels
+          path: dist
+
+      - uses: pypa/gh-action-pypi-publish@master
+        with:
+          user: __token__
+          password: ${{ secrets.pypi_password }}
```
Comment (joshmoore) [inline .github/workflows/releases.yml:71]: I've added a PYPI_PASSWORD token to the repository.

**Comment by Carreau** [comment]
> Do we need to do anything special to indicate the wheel is not architecture dependent?

not as far as I can tell, I think that's one of the main difference between `build` and `cibuildwheel`. 
on a non-tagged repository I get 

```
$ ls -1 dist/
zarr-2.4.1.dev69+dirty-py3-none-any.whl
zarr-2.4.1.dev69+dirty.tar.gz
```

Indicating the wheel is compatible all platforms.

**Comment by Carreau** [comment]
Currently testing on myfork [https://github.com/Carreau/zarr-python/actions], the action does not work correctly yet.

**Review by rabernat — APPROVED** [review]

File: .github/workflows/releases.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      - uses: actions/checkout@v1
+        with:
+          submodules: true
+
+      - uses: actions/setup-python@v1
+        name: Install Python
+        with:
+          python-version: '3.8'
+
+      - name: Install PyBuil
```
Comment (jakirkham) [inline .github/workflows/releases.yml:26]: Small suggestion

```suggestion
      - name: Install PyBuild
```

**Review by jakirkham — APPROVED** [review]

File: .github/workflows/releases.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+          ls dist
+
+  upload_pypi:
+    needs: [build_artifacts]
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags/v')
+    steps:
+      - uses: pypa/gh-action-pypi-publish@master
+        with:
+          user: __token__
```
Comment (rabernat) [inline .github/workflows/releases.yml:59]: Just out of curiosity, what is this? In other projects, we use `{{ secrets.pypi_username }}`.

File: .github/workflows/releases.yml [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+jobs:
+  build_artifacts:
+    name: Build wheel on ubuntu-latest
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    env:
+      CIBW_TEST_COMMAND: pytest --pyargs zarr
+      CIBW_TEST_REQUIRES: pytest
+      CIBW_SKIP: "*27* pp*"
```
Comment (Carreau) [inline .github/workflows/releases.yml:14]:
```suggestion
```

File: .github/workflows/releases.yml (hunk 3 above)
Comment (Carreau) [reply .github/workflows/releases.yml:55]: PyPI let you generate tokens for single projects. https://pypi.org/help/#apitoken

**Comment by jakirkham** [comment]
Thanks Matthias! 😄

**Pull request merged**